### DISCUSSION TITLE
Differentiate between calendar weeks and x times in week

### DISCUSSION THREAD

**Discussion by plainerman** [iSoron/uhabits#1714] [Feature Requests]
I am referring to https://github.com/iSoron/uhabits/issues/1578, where @hiqua said, that it might be nice to have calendar / "human weeks" instead of 7 arbitrary days. I think in the previous versions it made sense not to add a new option.

In the latest version, there is now an additional setting that allows you to set: x times per week, or x times in y days. The current behavior is that when you choose x times in 7 days, it changes to x times per week. I think this would be a good opportunity to differentiate between "human week" and x times in 7 days.

With this additional option, the old setting could just refer to x times in 7 days so we keep backwards compatibility.

A similar idea has also been proposed in https://github.com/iSoron/uhabits/discussions/975 

What do you think?

**Comment by hiqua** [comment]
Why not, but a lot of things would need to be reimplemented. Right now all the UI choices end up being two numbers describing the fraction of time the habit is done. This is a completely different model. I'm not sure the code would be clean enough to do that easily, but you're welcome to give it a try!

**Reply by plainerman** [reply]
Unfortunate that it might be tricky to implement. Could you point me to a few places in the code which might be a good starting point? 
Then I can see whether it would be feasible for me.

Thanks!

**Reply by hiqua** [reply]
I'd suggest to just start by running the app in debug mode, look at the various files, put breaking points to see what runs / what doesn't and familiarize yourself with the codebase. E.g. you can look for strings you see in the app and see where they appear in the codebase to map the UI to the code.

**Comment by RadekPilich** [comment]
What is a human week? A calendar week?

**Reply by plainerman** [reply]
Yes, I was referring to a calendar week. I have updated the post.
